src/main/java/com/sap/oss/phosphor/fosstars/data/artifact/VulnerabilitiesFromNpmAudit.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * This data provider tries to fill out the {@link OssFeatures#VULNERABILITIES_IN_ARTIFACT}. It
+ * gathers vulnerabilities from {@link NpmArtifact} using NPM Audit.
+ *
+ * @see <a href="https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities">NPM
+ * Audit</a>
+ */
+public class VulnerabilitiesFromNpmAudit implements DataProvider {
+
+  /**
+   * A logger.
+   */
+  private final Logger logger = LogManager.getLogger(getClass());
+
+  /**
+   * NPM registry URL to access security audit.
+   */
+  private static final String NPM_AUDIT_URL =
+      "https://registry.npmjs.org/-/npm/v1/security/audits";
+
+  /**
+   * NPM Audit API POST request body template.
+   */
+  private static final String
+      NPM_AUDIT_URL_BODY_ENTITY =
+      String.join("",
+          "{",
+          "\"name\": \"npm-vulnerabilities\",",
+          "\"version\": \"1.0.0\",",
+          "\"requires\" :{\"%s\": \"%s\"},",
+          "\"dependencies\": {\"%s\": {\"version\": \"%s\"}}",
+          "}");
+
+  /**
+   * An interface to NVD.
+   */
+  private final NVD nvd;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param nvd An {@link NVD}.
+   */
+  public VulnerabilitiesFromNpmAudit(NVD nvd) {
+    this.nvd = nvd;
+  }
+
+  /**
+   * The method always returns false, so that all child classes can't be interactive.
+   */
+  @Override
+  public final boolean interactive() {
+    return false;
+  }
+
+  /**
+   * This is a dummy cache which stores nothing.
+   */
+  @Override
+  public ValueCache<Subject> cache() {
+    return NoValueCache.create();
+  }
+
+  /**
+   * There is no call back required for this data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit set(UserCallback callback) {
+    return this;
+  }
+
+  /**
+   * No cache value is needed that is used by the data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit set(ValueCache<Subject> cache) {
+    return this;
+  }
+
+  /**
+   * No configuration is required for this data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit configure(Path config) throws IOException {
+    return this;
+  }
+
+  @Override
+  public VulnerabilitiesFromNpmAudit update(Subject subject, ValueSet values)
+      throws IOException {
+    logger.info("Gathering vulnerabilities from NPM Audit...");
+    Objects.requireNonNull(values, "On no! Values cannot be null");
+    NpmArtifact artifact = cast(subject, NpmArtifact.class);
+
+    if (!artifact.version().isPresent()) {
+      throw new IOException("Oh no! The version is not available.");
+    }
+
+    List<CVEMetadata> cves = cvesFrom(artifact);
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    for (CVEMetadata cveMetadata : cves) {
+      Optional<NvdEntry> nvdEntry = nvd.searchNvd(nvdEntryFrom(cveMetadata.cveID));
+      if (nvdEntry.isPresent()) {
+        vulnerabilities.add(Vulnerability
+            .Builder.from(nvdEntry.get()).set(cveMetadata.resolution).make());
```

Comment:
```suggestion
      nvdEntry.ifPresent(entry -> 
          vulnerabilities.add(
              Vulnerability.Builder.from(entry).set(cveMetadata.resolution).make()));
```

Suggestion based on personal preference 😉